src/omv/ports/stm32/omv_i2c.c

```diff
@@ -184,6 +184,7 @@ int omv_i2c_init(omv_i2c_t *i2c, uint32_t bus_id, uint32_t speed) {
         return -1;
     }
 
+    #if !defined(OMV_I2C_DISABLE_FASTMODE)
```

Comment:
```suggestion
    #if defined(IS_I2C_FASTMODEPLUS)
```

This should work, and you won't need to define new options.

Reply:
> It seems like we could still run into the poisonous cast in the preprocessor; should that be a runtime check?

This only tests if the macro is defined, which is true if FMP is supported, otherwise it's not defined at all.
```
#if defined(IS_I2C_FASTMODEPLUS)
if (speed == OMV_I2C_SPEED_FAST) {
....
```